File: 118-Pascal's-Triangle/solution.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> res;
        res.resize(numRows);
        
        for (int i = 0; i < numRows; ++i) {
            res[i].resize(i + 1);
            res[i][0] = 1;
            res[i][res[i].size() - 1] = 1;
            for (int j = 1; j < res[i].size() - 1; j++) {
                res[i][j] = res[i - 1][j - 1] + res[i - 1][j];
            }
        }
        return res;
    }
};

File: 118-Pascal's-Triangle/solution.go
class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        result = []
        for i in xrange(numRows):
            result.append([])
            for j in xrange(i + 1):
                if j in (0, i):
                    result[i].append(1)
                else:
                    result[i].append(result[i - 1][j - 1] + result[i - 1][j])
        return result